ui/ducks/metamask/metamask.js

```diff
@@ -307,7 +308,10 @@ export function getConversionRate(state) {
 }
 
 export function getNativeCurrency(state) {
-  return state.metamask.nativeCurrency;
```

Comment:
@jpuri @NiranjanaBinoy I updated this since your last review. The problem was pointed out to me by Mark. `state.metamask.nativeCurrency` is always set after the conversionRate updates. So if we stopped using it, the `nativeCurrency` and `conversionRate` could get out of sync, showing the user incorrect data.

I have changed the code so that we still use the `nativeCurrency`, but only when the useCurrencyRateCheck setting is on. When it is off, which is required for the bug we are trying to fix, we don't show converted values, so the concern about `nativeCurrency` and `conversionRate` being out of sync does not matter

Reply:
To add more detail: currently we continue showing the _old_ network currency and currency rate after switching, until the new rate has been loaded. This is pretty confusing, but we're using the CurrencyRateController `nativeCurrency` state in this situation so we can at least be confident the symbol we're showing next to the rate matches the rate.

Ideally we would show _no_ fiat value while currency rates are loading. That would be less confusing. Once we do that, I think it would be safe to always use `state.metamask.provider.ticker` for the current currency.

But until we make that improvement, there is this tradeoff to consider.